src/nexus.cpp

```diff
@@ -160,31 +152,35 @@ void Nexus::start()
  */
 void Nexus::showLogin()
 {
+    loginScreen = new LoginScreen();
+
+#ifdef Q_OS_MAC
+    connect(loginScreen, &LoginScreen::windowStateChanged, this,
+            &Nexus::onWindowStateChanged);
+#endif
+
     delete widget;
     widget = nullptr;
 
     delete mProfile;
     mProfile = nullptr;
 
     loginScreen->reset();
-    loginScreen->move(QApplication::desktop()->screen()->rect().center() - loginScreen->rect().center());
     loginScreen->show();
-    ((QApplication*)qApp)->setQuitOnLastWindowClosed(true);
+    loginScreen->move(QApplication::desktop()->screen()->rect().center() -
```

Comment:
I think, it should be moved before show? It will prevent showing in wrong place on few second if system will produse some freeze